src/edges_analysis/analysis/filters.py

```diff
@@ -998,3 +998,51 @@ def negative_power_filter(*, data: CalibratedData):
     These integrations obviously have some weird stuff going on.
     """
     return np.array([np.any(spec <= 0) for spec in data.spectrum])
+
+
+@step_filter(axis="time", data_type=(RawData, CalibratedData))
+def maxfm_filter(*, data: CalibratedData, threshold: float = 200):
+    """Max FM power filter.
+
+    This takes power of the spectrum between 80 MHz and 120 MHz(the fm range).
+    In that range, it checks each frequency bin to the estimated values..
+    using the mean from the side bins.
+    And then takes the max of all the all values that exceeded its expected..
+    value (from mean).
+    Compares the max exceeded power with the threshold and if it is greater
+    than the threshold given, the integration will be flagged.
+    """
+    fm_freq = (data.raw_frequencies >= 88) & (data.raw_frequencies <= 120)
+    # freq mask between 80 and 120 MHz for the FM range
+
+    if not np.any(fm_freq):
+        return np.zeros(len(data.spectrum), dtype=bool)
+
+    fm_power = data.spectrum[:, fm_freq]
+    maxfm = np.zeros_like(fm_power)
+
+    for i in range(len(data.spectrum)):
```

Comment:
There's a few things here: 

1. This is going to be slow (for loops are bad)
2. You're looping over the whole `data.spectrum` instead of `fm_power`

I'd probably do the averaging like this: `avg = (fm_power[2:] + fm_power[:-2])/2`. Then find the difference between the mid-bin and the avg by doing `diff = fm_power[1:-1] - avg`, then just take the max of that.